posthog/api/exports.py

```diff
@@ -0,0 +1,95 @@
+import json
+from typing import Any, Dict
+
+from django.core.serializers.json import DjangoJSONEncoder
+from django.http import HttpResponse
+from rest_framework import mixins, serializers, viewsets
+from rest_framework.authentication import BasicAuthentication, SessionAuthentication
+from rest_framework.decorators import action
+from rest_framework.exceptions import ValidationError
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.renderers import TemplateHTMLRenderer
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from posthog.api.dashboard import DashboardSerializer
+from posthog.api.insight import InsightSerializer
+from posthog.api.routing import StructuredViewSetMixin
+from posthog.auth import PersonalAPIKeyAuthentication
+from posthog.event_usage import report_user_action
+from posthog.models.exported_asset import ExportedAsset
+from posthog.permissions import ProjectMembershipNecessaryPermissions, TeamMemberAccessPermission
+from posthog.tasks import exporter
+
+
+class ExportedAssetSerializer(serializers.ModelSerializer):
+    """Standard ExportedAsset serializer that doesn't return content."""
+
+    class Meta:
+        model = ExportedAsset
+        fields = ["id", "dashboard", "insight", "export_format", "created_at", "has_content"]
+        read_only_fields = ["id", "created_at", "has_content"]
+
+    def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> ExportedAsset:
+        request = self.context["request"]
+        validated_data["team_id"] = self.context["team_id"]
+
+        if not validated_data.get("dashboard") and not validated_data.get("insight"):
+            raise ValidationError("Either dashboard or insight is required for an export")
```

Comment:
To make the API perhaps more obvious and make this `ValidationError` impossible, this could be a POST detail action `/export` on the insight/dashboard (that could be abstracted with a mixin), but this is RESTful too

Reply:
Yeah thought about it (that was actually the previous API design) but given some of the future changes as well I think this dedicated endpoint will make the most sense 👍